Library/Homebrew/tap.rb

```diff
@@ -712,10 +712,10 @@ def formula_files
       if formula_dir == path
         # We only want the top level here so we don't treat commands & casks as formulae.
         # Sharding is only supported in Formula/ and HomebrewFormula/.
-        formula_dir.children
+        Dir.glob(File.join(formula_dir.to_s, "*.rb"))
       else
-        formula_dir.find
-      end.select { formula_file?(_1) }
+        Dir.glob(File.join(formula_dir.to_s, "**/*.rb"))
```

Comment:
```suggestion
        formula_dir.glob("**/*.rb")
```
Is this also a lot slower?

Reply:
Yeah, that's still significantly slower.

```diff
diff --git a/Library/Homebrew/tap.rb b/Library/Homebrew/tap.rb
index 0fa277417d..39332b9319 100644
--- a/Library/Homebrew/tap.rb
+++ b/Library/Homebrew/tap.rb
@@ -712,10 +712,10 @@ class Tap
       if formula_dir == path
         # We only want the top level here so we don't treat commands & casks as formulae.
         # Sharding is only supported in Formula/ and HomebrewFormula/.
-        Dir.glob(File.join(formula_dir.to_s, "*.rb"))
+        formula_dir.glob("*.rb")
       else
-        Dir.glob(File.join(formula_dir.to_s, "**/*.rb"))
-      end.map { Pathname(_1) }
+        formula_dir.glob("**/*.rb")
+      end
     else
       []
     end
@@ -736,7 +736,7 @@ class Tap
   sig { returns(T::Array[Pathname]) }
   def cask_files
     @cask_files ||= if cask_dir.directory?
-      Dir.glob(File.join(cask_dir.to_s, "**/*.rb")).map { Pathname(_1) }
+      cask_dir.glob("**/*.rb")
     else
       []
     end
```

```console
$ hyperfine --parameter-list branch speed-up-loading-cask-and-formula-file-names_v2,speed-up-loading-cask-and-formula-file-names_v3 --warmup 5 --setup 'git switch {branch}' 'brew cat gimp'
Benchmark 1: brew cat gimp (branch = speed-up-loading-cask-and-formula-file-names_v2)
  Time (mean ± σ):      1.499 s ±  0.009 s    [User: 0.909 s, System: 0.553 s]
  Range (min … max):    1.488 s …  1.521 s    10 runs

Benchmark 2: brew cat gimp (branch = speed-up-loading-cask-and-formula-file-names_v3)
  Time (mean ± σ):      1.721 s ±  0.010 s    [User: 1.134 s, System: 0.552 s]
  Range (min … max):    1.707 s …  1.737 s    10 runs

Summary
  brew cat gimp (branch = speed-up-loading-cask-and-formula-file-names_v2) ran
    1.15 ± 0.01 times faster than brew cat gimp (branch = speed-up-loading-cask-and-formula-file-names_v3)
```